Java/other/MergeSortedArrays.java

```diff
@@ -0,0 +1,76 @@
+import java.util.*;
+
+class MergeSortedArrays
+{
+	private static void merge(int arr1[], int arr2[])
+	{
+		// Iterate through all elements of arr2[] in reverse 
+		for (int i = (arr2.length - 1); i >= 0; i--)
+		{
+            // In here we compare the different terms of
+            // both arrays and place them in respective positions
+			int j, last = arr1[arr1.length - 1];
+			for (j = arr1.length - 2; j >= 0 && arr1[j] > arr2[i]; j--) {
+                arr1[j+1] = arr1[j];
+            }
+			// If there was a greater element
+			if (j != arr1.length - 2 || last > arr2[i])
+			{
+				arr1[j+1] = arr2[i];
+				arr2[i] = last;
+			}
+		}
+	}
+
+	public static void main(String[] args)
+	{
+        Scanner sc = new Scanner(System.in);
+
+        // taking input array 1
+        System.out.println("Enter size of first array:");
+        int size1 = sc.nextInt();
+        int arr1[] = new int[size1];
+        System.out.println("Enter array elements:");
+        for (int i=0; i<size1; i++) {
+            arr1[i] = sc.nextInt();
+        }
+        // taking input array 2
+        System.out.println("Enter size of second array:");
+        int size2 = sc.nextInt();
+        int arr2[] = new int[size2];
+        System.out.println("Enter array elements:");
+        for (int i=0; i<size2; i++) {
+            arr2[i] = sc.nextInt();
+        }
+        sc.close();
+        
+        merge(arr1, arr2);
+
+        System.out.println("Sorted arrays are:");
+        for (int i = 0; i < size1; i++) {
+            System.out.print(arr1[i] + " ");
+        }
+        System.out.println();
+        for (int i = 0; i < size2; i++) {
+            System.out.print(arr2[i] + " ");
+        }
+	}
+}
+/**
+ * Sample input/output
+ * Enter size of first array:
+ * 6
+ * Enter array elements:
+ * 1 8 9 15 16 18
+ * Enter size of second array:
+ * 4
+ * Enter array elements:
+ * 2 4 7 10
+ * Sorted arrays are:
+ * 1 2 4 7 8 9 
+ * 10 15 16 18
+ * 
+ * Time complexity: O(m*n)
```

Comment:
I am a little confused about this sir.
We have nested for loops in the function, so shouldn't it be (m*n)? Because for every run of the outer loop the inner loop will run once.

Reply:
@ritvij14 Yeah you are correct what I was referring was to used better approach like the time complexity can be improved to O(m+n) using something this 
```
private static void merge(int arr1[], int size2, int arr2[], int size1) {
    if (arr1 == null || arr2 == null || size2 == 0) {
        return;
    }

    int i = size2 - 1;
    int j = size1 - 1;
    int k = size2 + size1 - 1;

    while (i >= 0 && j >= 0) {
        if (arr1[i] >= arr2[j]) {
            arr1[k--] = arr1[i--];
        } else {
            arr1[k--] = arr2[j--];
        }
    }

    while (j >= 0) {
        arr1[k--] = arr2[j--];
    }
}
```
Your code works fine this is just an optional approach for better time complexity.

Reply:
@iamrajiv thanks for telling this method, I didn't think this way :sweat_smile:. I will make changes. This definitely looks better thanthe previous time complexity.

Reply:
@iamrajiv Ok got it. Just one doubt though, `arr1[k--]` will give `OutOfBoundsException`. How do we tackle that?